src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -12,40 +13,46 @@ namespace Enable.Extensions.Queuing.AzureServiceBus.Internal
     // TODO Respect cancellationToken
     public class AzureServiceBusQueueClient : BaseQueueClient
     {
-        private readonly string _connectionString;
-        private readonly string _queueName;
-        private readonly MessageReceiver _messageReceiver;
-        private readonly MessageSender _messageSender;
+        private static readonly ConcurrentDictionary<int, AzureServiceBusQueue> _queues
+            = new ConcurrentDictionary<int, AzureServiceBusQueue>();
 
+        private readonly int _queueKey;
+        private readonly AzureServiceBusQueue _queue;
         private bool _disposed;
 
         public AzureServiceBusQueueClient(
             string connectionString,
             string queueName)
         {
-            _connectionString = connectionString;
-            _queueName = queueName;
+            _queueKey = $"{connectionString}:{queueName}".GetHashCode();
 
-            _messageReceiver = new MessageReceiver(
-                    connectionString,
-                    queueName,
-                    ReceiveMode.PeekLock);
+            _queue = _queues.AddOrUpdate(
+                _queueKey,
+                new AzureServiceBusQueue(connectionString, queueName),
+                (_, queue) =>
+                {
+                    queue.AddReference();
+                    return queue;
+                });
+        }
 
-            _messageSender = new MessageSender(connectionString, queueName);
+        internal AzureServiceBusQueue Queue
+        {
+            get => _queue;
         }
 
         public override Task AbandonAsync(
             IQueueMessage message,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            return _messageReceiver.AbandonAsync(message.LeaseId);
+            return _queue.Receiver.AbandonAsync(message.LeaseId);
```

Comment:
Let's have a `private readonly` property called `_messageReceiver`, to save us having to look up this property throughout this class. Similarly, let's do the same for `_queue.Sender`, and have a property called `_messageSender`. This will also significantly reduce the number of changed lines in this PR.

Reply:
See comment [here](https://github.com/EnableSoftware/Enable.Extensions.Queuing/pull/28#issuecomment-471521347) and [latest code](666dab50be8bc06913c2ddc9127499345b6cbbed).